### DISCUSSION TITLE
unstable results with dev version

### DISCUSSION THREAD

**Discussion by ehsandeep** [projectdiscovery/dnsx#304] [Q&A]
### dnsx version:
dev

### Steps To Reproduce:

Running with **latest** version:

```console
while true; do echo total: `chaos -d hackerone.com | dnsx -silent | wc`; done

total: 10
total: 10
total: 10
total: 11
total: 10
total: 11
total: 10
total: 11
```

Running with **dev** version:

```console
while true; do echo total: `chaos -d hackerone.com | ./dnsx -silent | wc`; done

total: 7
total: 8
total: 9
total: 8
total: 10
total: 9
total: 9
total: 9
total: 9
```

### Expected Behavior:
stable / expected results with dev code.

**Comment by parrasajad** [comment]
not able to replicate the issue, getting stable results over multiple runs on `dev`
```console
while true; do echo total: `chaos -d hackerone.com -silent | ./dnsx -silent | wc -l`; done

total: 11
total: 11
total: 11
total: 11
total: 11
total: 11
total: 11
total: 11
total: 11
...
```

**Comment by Mzack9999** [comment]
The difference in enumeration depends on which resolver is used during iterations, as they are rotated in a round-robin fashion. A -/+ 1 item difference among runs with UDP protocol is acceptable. Shall we attempt to resolve with all resolvers at least once until one is responsive? cc @ehsandeep

**Comment by ehsandeep** [comment]
Moving to discussions as we are unable to repro this!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
